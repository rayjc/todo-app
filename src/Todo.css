.Todo {
    margin: 10px 20px;
}

.completed {
    text-decoration: line-through;
    color: grey;
}

.Todo-btn {
    margin: auto 10px;
    border: none;
}