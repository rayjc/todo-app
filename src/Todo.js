import React, { useState } from 'react';
import './Todo.css';


const Todo = ({ id, name, isDone, removeTodo }) => {
  const [isChecked, setIsChecked] = useState(isDone);

  return (
    <div className="Todo">
      <input type="checkbox" checked={isChecked} onChange={(e) => setIsChecked(e.target.checked)} />
      <span className={isChecked ? "completed" : ""}>{name}</span>
      {isChecked && <button className="Todo-btn" onClick={() => removeTodo(id)}>x</button>}
    </div>
  )
}


export default Todo;