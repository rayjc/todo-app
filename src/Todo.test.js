import React from "react";
import { render } from "@testing-library/react";
import Todo from "./Todo";

it("renders without crashing", function() {
  render(<Todo />);
});

it("should match default snapshot", function() {
  const { asFragment } = render(
    <Todo id="0" name="test" isDone={false} removeTodo={jest.fn()} />
  );
  expect(asFragment()).toMatchSnapshot();
})