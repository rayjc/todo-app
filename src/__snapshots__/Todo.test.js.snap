// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match default snapshot 1`] = `
<DocumentFragment>
  <div
    class="Todo"
  >
    <input
      type="checkbox"
    />
    <span
      class=""
    >
      test
    </span>
  </div>
</DocumentFragment>
`;
